chia/wallet/trading/trade_store.py

```diff
@@ -251,38 +327,112 @@ async def get_all_trades(self) -> List[TradeRecord]:
         return records
 
     async def get_trades_between(
-        self, start: int, end: int, sort_key: Optional[str] = None, reverse: bool = False
+        self,
+        start: int,
+        end: int,
+        *,
+        sort_key: Optional[str] = None,
+        reverse: bool = False,
+        exclude_my_offers: bool = False,
+        exclude_taken_offers: bool = False,
+        include_completed: bool = False,
     ) -> List[TradeRecord]:
         """
         Return a list of trades sorted by a key and between a start and end index.
         """
-        records = await self.get_all_trades()
+        if start < 0:
+            raise ValueError("start must be >= 0")
 
-        # Sort
-        records = sorted(records, key=attrgetter("trade_id"))  # For determinism
+        if start > end:
+            raise ValueError("start must be less than or equal to end")
+
+        # If excluding everything, return an empty list
+        if exclude_my_offers and exclude_taken_offers:
+            return []
+
+        offset = start
+        limit = end - start
+        where_status_clause: Optional[str] = None
+        order_by_clause: Optional[str] = None
+
+        if not include_completed:
+            # Construct a WHERE clause that only looks at active/pending statuses
+            where_status_clause = (
+                f"(status={TradeStatus.PENDING_ACCEPT.value} OR "
+                f"status={TradeStatus.PENDING_CONFIRM.value} OR "
+                f"status={TradeStatus.PENDING_CANCEL.value}) "
+            )
+
+        # Create an ORDER BY clause according to the desired sort type
         if sort_key is None or sort_key == "CONFIRMED_AT_HEIGHT":
-            records = sorted(records, key=attrgetter("confirmed_at_index"), reverse=(not reverse))
+            order_by_clause = (
+                f"ORDER BY confirmed_at_index {'ASC' if reverse else 'DESC'}, "
+                f"trade_id {'DESC' if reverse else 'ASC'} "
+            )
         elif sort_key == "RELEVANCE":
```

Comment:
The reason these values were put into a class and a separate file was so that we could avoid adding an elif everytime we have something new to sort by.  Maybe extracting this functionality to a separate class or file would be could and would make this now massive function much more readable.